src/Types/ContractInitializedEvent.cs

```diff
@@ -0,0 +1,51 @@
+using Concordium.Sdk.Exceptions;
+
+namespace Concordium.Sdk.Types;
+
+/// <summary>
+/// Data generated as part of initializing a single contract instance.
+/// </summary>
+public class ContractInitializedEvent
+{
+    /// <summary>
+    /// Contract version.
+    /// </summary>
+    public ContractVersion ContractVersion { get; init; }
+    /// <summary>
+    /// A reference to a smart contract module deployed on the chain.
+    /// </summary>
+    public ModuleReference ModuleReference { get; init; }
+    /// <summary>
+    /// Represents a contract address.
+    /// </summary>
+    public ContractAddress ContractAddress { get; init; }
+    /// <summary>
+    /// Represents a CCD amount.
+    /// </summary>
+    public CcdAmount Amount { get; init; }
+    /// <summary>
+    /// A contract name.
+    /// </summary>
+    public OwnedContractName InitName { get; init; }
+    /// <summary>
+    /// A list of events generated by a smart contract.
+    /// </summary>
+    public IList<ContractEvent> Events { get; init; }
+
+    internal ContractInitializedEvent(Concordium.Grpc.V2.ContractInitializedEvent initializedEvent)
```

Comment:
When I look at it why are we using`From`? To me it feels like we are copying Rust style.

Reply:
Some subscribe to not having logic and throwing exceptions in constructors, and at least that's what I did to the largest possible extent in the codebase. I think it would be nice if we were consistent.

As for the reasoning for my choice of that, `From` does checks that throw exceptions, and while you would expect the parameters of the constructors to be stored, this assumption goes away when using `From`, so it provides for some further encapsulation in general.